### PR TITLE
Notnine 3.0

### PR DESCRIPTION
**Pull request by notnine**
Update 3.0
Changes Made

Added unit tests:
UserManagerTest
UserTest

Decreased smell in:
UserProfilePresenter
UserManager

====================================================
Completed Changes

Option to return to main menu
Actions: Following a user, unfollowing a user, browsing a user, viewing a user's posts using the PostController
Javadocs for all methods

### CODE REVIEW

File: src/main/java/controllers/UserProfileController.java [hunk 1]
```diff
@@ -87,128 +90,164 @@ else if (choice == 1) {
...
+        this.inOut.setOutput("Currently registered users:" + allUsernames.substring(0, allUsernames.length() - 2));
+    }
+
+    /**
+     * Display user's posts
+     */
+    private void runDisplayUserPosts(User user) {
+        PostController postController = new PostController(this.inOut, this.databaseManager, this.loginManager);
+        PostManager postManager = new PostManager(this.userManager.getUserPosts(user));
+        for(Post post: this.userManager.getUserPosts(user)) {
```
Comment (ericjest) [inline src/main/java/controllers/UserProfileController.java:212]: Could we add a space after "for" for style consistency?

File: src/main/java/use_cases/UserManager.java [hunk 2]
```diff
@@ -2,176 +2,199 @@
...
+            updatedFollowingList.addAll(currentUser.getFollowing());
             currentUser.setFollowing(updatedFollowingList);
             return true;
         }
     }
 
     /**
-     * unfollow User want_to_unfollow, removing want_to_unfollow from the logged-in user's following list,
-     * return false if want_to_unfollow is not followed to begin with, true if action done successfully
+     * unfollow User want_to_unfollow, removing want_to_unfollow from the logged-in user's following list
```
Comment (ericjest) [inline src/main/java/use_cases/UserManager.java:102]: Could we rename the variable names in the comment to use camelCase?

Comment (ericjest) [reply src/main/java/use_cases/UserManager.java:102]: Or just put a space between

File: src/main/java/controllers/UserProfileController.java [hunk 3]
```diff
@@ -87,128 +90,164 @@ else if (choice == 1) {
...
+            postController.displayPost(postManager.getPostId(post));
+        }
+    }
+
+    /**
+     * Run the customize username logic
+     */
+    private void runCustomizeUsername() {
```
Comment (ericjest) [inline src/main/java/controllers/UserProfileController.java:220]: We’re basically just running runChangeUsernameDisplay in this method, maybe we could just get rid of this method and move the getInput into runChangeUsernameDisplay. Same thing with the password method. Is there a good reason we have this method?

**Review by ericjest — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -89,3 +89,4 @@ fabric.properties
 
 # Android studio 3.1+ serialized cache file
 .idea/caches/build_file_checksums.ser
+*.json
```
Comment (ericjest) [inline .gitignore:92]: Is there a specific reason you added this

File: src/main/java/controllers/UserProfileController.java [hunk 5]
```diff
@@ -87,128 +90,164 @@ else if (choice == 1) {
...
      */
-    private String displayFollowingList(User user) {
-        ArrayList<String> followingListUsernames = this.userManager.getFollowingListUsernames(user);
-        String followingList = "";
-        for (String username: followingListUsernames) {
-            followingList = followingList.concat(username + ", ");
+    private void runCustomizePassword() {
+        try {
+            String newPassword = this.inOut.getInput("Enter what you'd like your new password to be: ");
+            runChangePasswordDisplay(this.loginManager.getCurrUser(), newPassword);
```
Comment (ericjest) [inline src/main/java/controllers/UserProfileController.java:235]: Could we use this.methodName for any class methods?

**Review by ericjest — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/UserProfileController.java (hunk 1 above)
Comment (notnine) [reply src/main/java/controllers/UserProfileController.java:212]: yeah for sure

File: src/main/java/use_cases/UserManager.java (hunk 2 above)
Comment (notnine) [reply src/main/java/use_cases/UserManager.java:102]: yeah sounds good

File: src/main/java/controllers/UserProfileController.java (hunk 3 above)
Comment (notnine) [reply src/main/java/controllers/UserProfileController.java:220]: Iirc, I tried just running runChangeUsernameDisplay, but with this method, the function seemed cleaner and less smelly, although I'll revisit it to see which one is more readable

File: .gitignore (hunk 4 above)
Comment (notnine) [reply .gitignore:92]: oh no I dont know how this got there

File: src/main/java/controllers/UserProfileController.java (hunk 5 above)
Comment (notnine) [reply src/main/java/controllers/UserProfileController.java:235]: yeah I forgot to put this.runChangePasswordDisplay I think

File: src/main/java/controllers/UserProfileController.java (hunk 3 above)
Comment (ericjest) [reply src/main/java/controllers/UserProfileController.java:220]: What code smell does this solve?

File: .gitignore (hunk 4 above)
Comment (ericjest) [reply .gitignore]: Please remove then

File: src/main/java/controllers/UserProfileController.java (hunk 5 above)
Comment (ericjest) [reply src/main/java/controllers/UserProfileController.java]: It looks like this. is missing from a lot of your other method calls, please check

**Pull request merged**